pkg/sql/catalog/typedesc/type_desc.go

```diff
@@ -796,7 +796,6 @@ func EnsureTypeIsHydrated(
 				return err
 			}
 		}
-		return nil
```

Comment:
I think this was a bug that `EnsureTypeIsHydrated` didn't hydrate table implicit types which is actually a tuple. Let me know if this make sense..

Reply:
if we make this change, let's also add a test that shows why it's needed.

Reply:
ah yeah, good call, I'll add one for it.